src/Chapter3.hs

```diff
@@ -971,6 +1097,23 @@ implement the following functions:
 🕯 HINT: to implement this task, derive some standard typeclasses
 -}
 
+data DayOfWeek = Mon | Tue | Wed | Thu | Fri | Sat | Sun deriving (Show, Eq, Ord, Bounded, Enum)
+
+isWeekend :: DayOfWeek -> Bool
+isWeekend Sat = True
+isWeekend Sun = True
+isWeekend _   = False
+
+nextDay :: DayOfWeek -> DayOfWeek
+nextDay day
+    | day == maxBound = minBound
+    | otherwise       = succ day
```

Comment:
This is also a bit more maintainable. If you decide, for example, to have the "American" system of week days in the data type constructors, you won't need to change this function at all! 🙂